httpclient5/src/main/java/org/apache/hc/client5/http/psl/PublicSuffixMatcher.java

```diff
@@ -149,24 +148,30 @@ public String getDomainRoot(final String domain, final DomainType expectedType)
             }
             final DomainType domainRule = findEntry(rules, key);
             if (match(domainRule, expectedType)) {
-                if (domainRule == DomainType.PRIVATE) {
```

Comment:
@massdosage Unfortunately this change introduced a regression in 5.4.x. There is a reason for special treatment of `DomainType.PRIVATE`, we just did not have a test case for it in our test suite.

Reply:
Yes, I did say that this was a backwards-incompatible change as the previous behaviour didn't match the formal algorithm behaviour. We only focused on that aspect of it, if the `PublicSuffixListMatcher` is being used for other uses cases then I can't comment on their correctness as we're using this on the understanding that it implements the upstream standard. We're not using it for SSL/TLS etc. which I think are out of scope of the standard.

Reply:
@massdosage I did a bad job reviewing your changes so the fault is mine and mine only. The specification actually _does_ explicitly state that 
```
While some applications, such as browsers when considering cookie-setting, treat all entries the same, other applications may wish to treat ICANN domains and PRIVATE domains differently
```